apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/chat/ChatItemsLoader.kt

```diff
@@ -47,13 +49,19 @@ suspend fun apiLoadMessages(
         withChats {
           if (getChat(chat.id) == null) {
             addChat(chat)
+          } else if (replaceChat) {
+            replaceChat(chat.remoteHostId, chat.id, chat)
           } else {
             updateChatInfo(chat.remoteHostId, chat.chatInfo)
             updateChatStats(chat.remoteHostId, chat.id, chat.chatStats)
           }
         }
       }
       withChats(contentTag) {
+        if (redirectDisabled && chatModel.chatId.value != chat.id) {
```

Comment:
why do you need to have a check on chatId here if it's already checked before?
https://github.com/simplex-chat/simplex-chat/pull/5546/files#diff-c73fbf26887eb073849e49f529bfa56f8592ac068157bd63d7e963f2e5a7f73bR37

Reply:
not the same if you call this function with initial right? what was happening when I added artificial delay was this would redirect to the previous chat